src/makeRobot.js

```diff
@@ -38,7 +38,62 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name,
+    wheels,
+    version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, `
+      + `chip version: ${this.version}, `
+      + `wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    check(step) {
+      return step > 0 ? step : 0;
+    },
```

Comment:
I like it! But I would think of sth to make the name of this function more clear.
I would try sth like that:
```suggestion
    getValidSteps(step) {
      return step > 0 ? step : 0;
    },
```